Add configurable retry mechanism - #59 (#70)
SamVerschueren/dynongo: readme.md

});
```

#### Retry

> Retry configuration can be passed well during the initialisation. The retry mechanism is based on [p-retry](https://github.com/sindresorhus/p-retry) and requires the same [options](https://github.com/tim-kos/node-retry#retryoperationoptions). Configuring the retry will allow the user to automatically retry the dynongo operation if it's an retryable error.

Reviewer: ```suggestion
The retry configuration can be passed during initialisation, or per individual query. The mechanism is based on [`p-retry`](https://github.com/sindresorhus/p-retry) and requires the same [options](https://github.com/tim-kos/node-retry#retryoperationoptions). Configuring retry will allow the user to automatically retry the DynamoDB operation if it's a retryable error.
```

```js
db.connect({

> 	retryOptions: {

Reviewer: Maybe we should just go with `retry` here?

Author: Ok, will change it to a number as well and not the options from `p-retry`

### Methods

Every method can override the retry [options](https://github.com/tim-kos/node-retry#retryoperationoptions) passed with the `.connect()` method or can customise the retry configuration for the specific method.

```js

> Employee.find({Organisation: 'Amazon'}).where({Salary: {$gt: 3000}}).select('FirstName Name').retry({retries: 2}).exec()

Reviewer: ```suggestion
Employee
	.find({Organisation: 'Amazon'})
	.where({Salary: {$gt: 3000}})
	.select('FirstName Name')
	.retry({retries: 2})
	.exec()
```

### Methods

Every method can override the retry [options](https://github.com/tim-kos/node-retry#retryoperationoptions) passed with the `.connect()` method or can customise the retry configuration for the specific method.

```js
Employee
	.find({Organisation: 'Amazon'})
	.where({Salary: {$gt: 3000}})
	.select('FirstName Name')
	.retry({retries: 3, factor: 1, randomize: false})
	.exec()
	.then(employees => {
		// => [{FirstName: 'Foo', Name: 'Bar'}]
	});
```

If you don't want to configure the retry strategy, you can simply pass the number of retries.

```js
Employee
	.find({Organisation: 'Amazon'})
	.where({Salary: {$gt: 3000}})
	.select('FirstName Name')
	.retry(2)
	.exec()
	.then(employees => {

> 	// => [{FirstName: 'Foo', Name: 'Bar'}]

Reviewer: ```suggestion
		// => [{FirstName: 'Foo', Name: 'Bar'}]
```

### Methods

Every method can override the retry [options](https://github.com/tim-kos/node-retry#retryoperationoptions) passed with the `.connect()` method or can customise the retry configuration for the specific method.

```js
Employee
	.find({Organisation: 'Amazon'})
	.where({Salary: {$gt: 3000}})
	.select('FirstName Name')

> 	.retry({retries: 2})

Reviewer: Provide the same options as above to make it clear that it accepts all the p-retry stuff.

```suggestion
	.retry({retries: 3, factor: 1, randomize: false})
```

	.exec()
    .then(employees => {

>         // => [{FirstName: 'Foo', Name: 'Bar'}]

Reviewer: Indentation

Author: Could you give an suggestion on what the identation needs to be? It has the same identation as other documentation (see line 154)

Reviewer: Because of refined GitHub I noticed that it has spaces instead of tabs. Couldn't see the difference on the mobile app.

![image](https://user-images.githubusercontent.com/1913805/73597165-84127800-4529-11ea-8f44-4b61f746a7e5.png)

	});
```

If you don't want to configure the retry strategy, you can simply pass the number of retries.

```js
Employee
	.find({Organisation: 'Amazon'})
	.where({Salary: {$gt: 3000}})
	.select('FirstName Name')
	.retry(2)
	.exec()
    .then(employees => {
        // => [{FirstName: 'Foo', Name: 'Bar'}]

>     });

Reviewer: Indentation

Author: Could you give an suggestion on what the identation needs to be? It has the same identation as other documentation (see line 154)

Reviewer: It looked off when I reviewed it. On mobile now and it looks ok.

Reviewer: Same, it has spaces instead of tabs.
